geo/geomfn: implement ST_AsHexEWKB({geometry,text}), ST_AsHexEWKB({ge…

hueypark opened the pull request:
…ography,text})

Fixes #48880

Release note (sql change): Implemented the geometry based builtins `ST_AsHexEWKB` with encoding.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49644)
<!-- Reviewable:end -->

otan commented on pkg/sql/logictest/testdata/logic_test/geospatial:
```diff
@@ -249,25 +251,27 @@ SELECT
   ST_AsEWKB(geog),
   ST_AsHexWKB(geog),
   ST_AsHexEWKB(geog),
```
this output is getting a little harder to read. do you mind moving HexWKB/HexEWKB to a new query?

hueypark replied:
Done.

otan commented on pkg/sql/sem/builtins/geo_builtins.go:
```diff
@@ -695,6 +696,72 @@ var geoBuiltins = map[string]builtinDefinition{
+			},
+			Info: infoBuilder{
+				info: "Returns the EWKB representation in hex of a given Geometry. " +
+					"This variant has a second argument denoting the encoding.",
```
Do you mind adding what `xdr` and `ndr` means? The other builtin has this annotation.

hueypark replied:
Done.

hueypark committed:
geo/geomfn: implement ST_AsHexEWKB({geometry,text}), ST_AsHexEWKB({geography,text})

Fixes #48880

Release note (sql change): Implemented the geometry based builtins `ST_AsHexEWKB` with encoding.

hueypark commented:
Thank you for the review!

otan commented:
bors r+
